client/identity.go

```diff
@@ -21,26 +22,78 @@ type Identity struct {
 	LoomAddr       loom.Address
 }
 
-func CreateIdentity(name string, ethKey string, dappchainKey string, chainID string) (*Identity, error) {
-	mainnetPrivKey, err := crypto.HexToECDSA(strings.TrimPrefix(ethKey, "0x"))
-	if err != nil {
-		return nil, err
+func CreateRandomIdentity(chainID string) (*Identity, error) {
+    return CreateIdentity(nil, nil, chainID)
+}
+
+func CreateIdentity(hexKeyOrECDSA interface{}, signerOrKey interface{}, chainID string) (*Identity, error) {
+	var mainnetPrivKey *ecdsa.PrivateKey
```

Comment:
doesn't make sense why you need to have a private key, just to extract the public key

Reply:
Agreed

Reply:
You need the private key for example when you create identities that require mainnet interaction such as in the case of the gateways: https://github.com/loomnetwork/go-loom/blob/master/client/gateway/mainnet_gateway_client.go#L31 which calls https://github.com/loomnetwork/go-loom/blob/master/client/helpers.go#L23